src/main/kotlin/de/amplimind/codingchallenge/service/UserService.kt

```diff
@@ -92,27 +93,32 @@ class UserService(
      * @return the [UserInfoDTO] of the deleted user
      */
     fun deleteUserByEmail(email: String): UserInfoDTO {
-        // Check if the user is trying to delete himself
-        val auth = SecurityContextHolder.getContext().authentication
-        val authenticatedUserEmail = auth?.name
-        if (authenticatedUserEmail == email) {
-            throw UserSelfDeleteException("User with email $email cannot delete himself")
-        }
+        try {
+            // Check if the user is trying to delete himself
+            val auth = SecurityContextHolder.getContext().authentication
+            val authenticatedUserEmail = auth?.name
+            if (authenticatedUserEmail == email) {
+                throw UserSelfDeleteException("User with email $email cannot delete himself")
+            }
 
-        // Delete the submissions of the user
-        val submissions = this.submissionRepository.findByUserEmail(email)
-        if (submissions != null) {
-            this.submissionRepository.delete(submissions)
-        }
+            // Delete the submissions of the user
+            val submissions = this.submissionRepository.findByUserEmail(email)
+            if (submissions != null) {
+                this.submissionRepository.delete(submissions)
+            }
 
-        // Find the user & delete the user
-        val user = this.userRepository.findByEmail(email) ?: throw ResourceNotFoundException("User with email $email was not found")
-        this.userRepository.delete(user)
-        return UserInfoDTO(
-            email = user.email,
-            isAdmin = user.role.matchesAny(UserRole.ADMIN),
-            status = UserStatus.DELETED,
-        )
+            // Find the user & delete the user
+            val user = this.userRepository.findByEmail(email)
+                    ?: throw ResourceNotFoundException("User with email $email was not found")
+            this.userRepository.delete(user)
+            return UserInfoDTO(
+                    email = user.email,
+                    isAdmin = user.role.matchesAny(UserRole.ADMIN),
+                    status = UserStatus.DELETED,
+            )
+        } catch (ex: OptimisticLockingFailureException) {
+            throw IllegalStateException("The user was updated by another transaction")
```

Comment:
Des try catch kann fürs erste mal weg, um des kümmern wir uns wenn wir dieses retry von spring boot verwenden (aber dann halt in nen anderen ticket)^^